src/participant/prompts/query.ts

```diff
@@ -58,11 +62,17 @@ db.getCollection('');\n\n`;
     if (schema) {
       prompt += `Collection schema: ${schema}\n`;
     }
-    if (
-      sampleDocuments &&
-      prompt.length + sampleDocuments.length < MAX_TOTAL_PROMPT_LENGTH
-    ) {
-      prompt += `Sample socuments: ${sampleDocuments}\n`;
+    if (sampleDocuments) {
+      let docs = toJSString(sampleDocuments);
+      // First check the length of all stringified sample documents.
+      // If the resulting prompt is too large, proceed with only 1 sample document.
+      if (prompt.length + docs.length > maxInputTokens) {
+        docs = toJSString([sampleDocuments[0]]);
```

Comment:
In this case should we change the prompt addition to be
`Sample document` without the `s`? Also we can remove the array wrapper `[]`.

Reply:
Hmm not sure it makes much difference for the model. You can experiment as part of [VSCODE-583](https://jira.mongodb.org/browse/VSCODE-583) if this gives better results.

Reply:
Simplifying the prompt is a good practice where we can, the extra [] gives more for the ai to parse and reason about.

Reply:
I can do it in this pr! Just wanted to be sure this is really something that makes a difference. I was thinking that knowing that there is one, but there can be more documents in the collection gives the model a better impression about the dataset.